### ISSUE TITLE
Error when intercepting a fun with an atom literal in the fun head

### ISSUE THREAD

**Issue by mtarnovan** [amalbuquerque/interceptor#9]
When intercepting a fun like this:
```
def foo(bla, :bar) do
...
```

`Interceptor` crashes.

Here's a partial stacktrace that I think pinpoints the issue pretty well:

```
* (FunctionClauseError) no function clause matching in Interceptor.FunctionArguments.get_arg_name_and_its_ast/1    
    
    The following arguments were given to Interceptor.FunctionArguments.get_arg_name_and_its_ast/1:
    
        # 1
        :monthly
    
    Attempted function clauses (showing 5 out of 5):
    
        defp get_arg_name_and_its_ast({:=, _, [_operand_a, _operand_b] = assignment_operands} = arg_full_ast)
        defp get_arg_name_and_its_ast({:\\, _metadata, [arg_ast, _default_value_ast]} = arg_full_ast)
        defp get_arg_name_and_its_ast({arg_name, _metadata, _context} = arg_ast) when arg_name === :<<>> or arg_name === :{} or arg_name === :%{} or arg_name === :% or arg_name === :<>
        defp get_arg_name_and_its_ast({arg_name, _metadata, _context} = arg_ast) when is_atom(arg_name)
        defp get_arg_name_and_its_ast(arg_ast) when is_list(arg_ast) or is_tuple(arg_ast) or is_integer(arg_ast) or is_binary(arg_ast)
    
    (interceptor 0.5.3) lib/function_arguments.ex:119: Interceptor.FunctionArguments.get_arg_name_and_its_ast/1
    ```

**Comment by amalbuquerque** [comment]
@mtarnovan Thanks for reporting this. Will submit a fix soon.

**Comment by amalbuquerque** [comment]
@mtarnovan this is already fixed since Dec 21, 2019 (:exclamation:) but I forgot to release a new version :man_facepalming:
Plz check this commit: https://github.com/amalbuquerque/interceptor/commit/6b6d2613aa2fc8f2a684d53c5fcccef695cbebc9

I will now release a new `0.5.4` version with the fix and will close this issue. Plz bump your `interceptor` version to `0.5.4`.

@filipecabaco this is already fixed after all :sweat_smile:.

**Comment by amalbuquerque** [comment]
@mtarnovan just published version `0.5.4`: https://hex.pm/packages/interceptor/0.5.4

The error shouldn't happen anymore.
